indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,534 @@
+            var latestRoomValue = roomInput.val();
+
+            usingDefault.on('click', function() {
+                toggleDefaultLocation($(this).prop('checked'));
+            });
+
+            toggleDefaultLocation(usingDefault.prop('checked'), true);
+
+            venueInput.typeahead({
+                source: venues,
+                minLength: 0,
+                searchOnFocus: true,
+                resultContainer: '#' + venueInput.siblings('.typeahead-results').find('.results-list-container').attr('id'),
+                template: {% raw %}'<span id="location-venue-{{id}}">{{name}}</span>'{% endraw %},
+                emptyTemplate: function(query) { return ""; },
+                display: 'name',
+                callback: {
+                    onInit: function(node) {
+                        this.query = node.val();
+                    },
+                    onClickBefore: function(node, a, item, event) {
+                        validVenue = true;
+                    },
+                    onClickAfter: function(node, a, item, event) {
+                        venueInput.trigger('typeahead:click-on-option', {
+                            item_name: item.name,
+                            venue_id: item.id
+                        });
+                    },
+                    onLayoutBuiltBefore: function(node, query, result, resultHtmlList) {
+                        return updateResultsList(node, resultHtmlList);
+                    },
+                    onSearch: function(node, query) {
+                        this.query = cleanQuery(query);
+                    },
+                    onShowLayout: function(node, query) {
+                        venueInput.next('.highlighter').removeClass('visible');
+                        currentScrollPos = 0;
+                    },
+                    onHideLayout: function(node, query) {
+                        if (justUseOptionSelected) {
+                            venueInput.trigger('typeahead:click-on-custom-option', {item_name: query});
+                        } else if (!validVenue && node.val() && !preventGuessing) {
+                            var result = [];
+                            result = $.grep(venues.data, function(venue) {
+                                return venue.name.toUpperCase() == query.toUpperCase();
+                            });
+                            if (result.length > 0) {
+                                $('#location-venue-' + result[0].id).parent().trigger('click');
+                            } else {
+                                venueInput.trigger('typeahead:click-on-custom-option', {item_name: query});
+                            }
+                        }
+                        preventGuessing = false
+                        validVenue = false;
+                    },
+                    onNavigateBefore: function(node, query, event) {
+                        if (event.keyCode == 9) {
```

Comment:
It's in the legacy parts of our JS (maybe we should move it into a less smelly place?) so not knowing about it is not that weird :P